### PR TITLE
Api

### PR DESCRIPTION
**Pull request by chukwuemeka1234**
**Hi there** 👋 

### In this PR, the following was completed

- [ ] Create an API endpoint to list all posts for a user.

- [ ] Create an API endpoint to list all comments for a user's post.

- [ ] Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.

- [ ] Add authentication to your API endpoints.

- [ ] Your API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -11,4 +11,14 @@
...
+
+  namespace :api, defaults: { format: :json } do
+    namespace :v1 do
+      resources :users, only: %i[index show ] do
+        resources :posts, only: %i[index] do
+          resources :comments, only: %i[index create]
+        end
+      end
+    end
+  end
```
Comment (hiromataba) [inline config/routes.rb]: - [ ] In addition as the requirements says you need to add an API endpoint for authentication, both registration and login
* Login endpoint - **POST :** `/log in`
* Registration endpoint - **POST :** `/register` or `/signup`

File: app/controllers/api/v1/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.create(text: comment_parameters[:text], author_id: current_user.id)
+
+    if @comment.save
+      render json: @comment, status: 200
+    else
+      render json: @comment.errors, status: :bad_request, message: 'Failed to add comment'
+    end
+  end
```
Comment (hiromataba) [inline app/controllers/api/v1/comments_controller.rb]: - [ ] Kindly fix this error when a new comment is created.

image

**Review by hiromataba — CHANGES_REQUESTED** [review]
## Hi @team @chukwuemeka1234  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/api/v1/comments_controller.rb (hunk 2 above)
Comment (chukwuemeka1234) [reply app/controllers/api/v1/comments_controller.rb]: Working on it right now.

Comment (chukwuemeka1234) [reply app/controllers/api/v1/comments_controller.rb]: Taking more time than required to fix.

Comment (cwaku) [reply app/controllers/api/v1/comments_controller.rb]: All fixed now

**Review by mmsesay — APPROVED** [review]
Hi @chukwuemeka1234 & @cwaku,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**